### PR TITLE
Fixed the test cases

### LINKED ISSUE #111: Test Cases: Fixed
The current test cases are not written according to the new features which have been recently added. Assign me to write the test cases under gssoc-ext.

### PR DESCRIPTION
**Pull request by tannuiscoding**
### Title of the Pull Request
- [x] Have you renamed the PR Title in a meaningful way?

### Related Issue

closes #111

### Description

I changed the test cases of `models.py` and `piechart.py` according to the new features and they are running perfectly fine. 

I did not work on the test cases of `graph.py` ass those test cases and features are still left to be fixed.

test cases results for  `model.py`

Screenshot 2024-10-12 175228

test cases result `graph.py`

image

### Type of change

What sort of changes have you made:

```python
class MockDateTime(datetime):
    @classmethod
    def now(cls):
        return datetime.strptime("1234", "%H%M")

def mock_copyfile(src, dst):
    pass  

def mock_glob_single(pattern):
    return ["stash/test.py"]

def mock_glob_none(pattern):
    return []

def mock_glob_multiple(pattern):
    return ["stash/test.py", "stash/test2.py"]
```

All the test cases are working good and the issues are fixed. please review them and let me know if any changes are required.

Thank you!!

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist

- [x] My code follows the code style of this project.
- [x] I have followed the contribution guidelines
- [x] I have performed a self-review of my own code.
- [x] I have ensured my changes don't generate any new warnings or errors.
- [x] I have updated the documentation (if necessary).
- [x] I have resolved all merge conflicts.

### CODE REVIEW

File: tests/test_models.py
Comment (ShashwatAgrawal20) [inline tests/test_models.py:1]: The file is not using `os, shutil and subprocess` modules we can get rid of those import statements.

File: tests/test_piechart.py [hunk 1]
```diff
@@ -1,13 +1,62 @@
 import os
 import pytest
```
Comment (ShashwatAgrawal20) [inline tests/test_piechart.py:2]: We can get rid of these 2 in this files as we are not using them.

**Review by ShashwatAgrawal20 — CHANGES_REQUESTED** [review]
**MAKE SURE YOU'RE FORMATTING YOUR CODE.**


And just a quick-fix for the file `tests/test_graph.py` as this PR is related to test cases it can be get fixed too. 

1. You can use a single import statement for importing both `display and copy_to_clipboard` something like `from package.graph import copy_to_clipboard, display` currently the code uses 2 different import statement.
2. we don't need these imports there `os, sys and shutil` as we aren't using those in that file. 
3. That assert statement should be something like `assert display("test",current_time) is None` instead of using `==`
4. And the last except statement should be a f-string as it complains about e is assigned but never used. `pytest.fail(f"Unexpected error raised: {e}")`

File: tests/test_models.py
Comment (this-is-yaash) [reply tests/test_models.py:1]: we don't need file i/o operations for this models.py
It needs to have
- print
- clipboard copy

**Review by this-is-yaash — APPROVED** [review]
LGTM!

**Review by jindalpriyanshu101 — APPROVED** [review]
LGTM!, Approved.

**Comment by ShashwatAgrawal20** [comment]
LGTM!, Approved!

shipit

**Comment by ShashwatAgrawal20** [comment]
LGTM! 

shipit

**Review by ShashwatAgrawal20 — APPROVED** [review]
LGTM! 

shipit

**Pull request merged**